### ISSUE TITLE
Guide for implementation

### ISSUE THREAD

**Issue by reynaldiRepo** [miketaylr/partitioned-storage-deprecation-trial-feedback#6]
@miketaylr 

I have 2 websites and 1 extension:

With the partitioned-storage-deprecation-trial, I want to achieve the following scenarios:

1. Website a.com is accessed by the user as the top-level website, and the website will write records to indexedDB on a.com. After that, the user is redirected to b.com.

2. On b.com, the user will see an extension that injects a script to render a.com inside an iframe.

I want the indexedDB data written on a.com in the first process to be read on a.com when it's rendered inside the iframe.

I need guidance on how to set up a token (which origin I should use when registering the token), and whether the meta can only be injected from a JavaScript script.

As additional information, there are several instances like b.com rendering a.com in an iframe. Could you please explain how we can set up an origin trial token with the scenario I mentioned?

**Comment by miketaylr** [comment]
Hi @reynaldiRepo, have you read through https://developer.chrome.com/blog/storage-partitioning-deprecation-trial/ and https://developer.chrome.com/en/docs/web-platform/third-party-origin-trials/?



> 2\. On b.com, the user will see an extension that injects a script to render a.com inside an iframe.

Can you clarify what "the user will see an extension" means? Do the users already have this extension installed? Does the extension have host permissions for a.com or (and?) b.com?

**Comment by reynaldiRepo** [comment]
@miketaylr Thank you for responding to my question.

Yes, I have read https://developer.chrome.com/blog/storage-partitioning-deprecation-trial/ and https://developer.chrome.com/en/docs/web-platform/third-party-origin-trials/, and I have tried to implement them.

However, I have not yet found the behavior I desire. The indexedDB from a.com, rendered in an iframe, is still isolated due to partitioning.

Furthermore, I am still not sure why the origin trials token I inject always shows the wrong origin. I am injecting from the extension to both b.com and a.com, where both tokens are different because of different origins (is my implementation incorrect?).

<img width="447" alt="image" src="https://github.com/miketaylr/partitioned-storage-deprecation-trial-feedback/assets/38524172/b65b5c72-e6cc-4a4e-b413-6d8e4aed2e59">

[image from dev tools heyfoodlama.com]


As for your questions:
1. Do the users already have this extension installed?
   Yes, the extension is already installed.

2. Does the extension have host permissions for a.com or (and?) b.com?
   Yes, we have permissions for both domains.

**Comment by arichiv** [comment]
What is the source (url) of the script injected into origin b? In order for a third-party OT token to work it must be (1) for the targeted origin (in this case origin a) and (2) injected by a script from that origin (in this case origin a).

**Comment by reynaldiRepo** [comment]
@arichiv Thank you for providing response.

Yes, we have already provided two tokens for both of them, but we have resolved this issue.

However, we do not know why we cannot obtain a valid token when our Third-party matching checkbox is enabled, even though the token for the top-level site is injected from the extension.

We also missed one step from this documentation https://developer.chrome.com/blog/storage-partitioning-deprecation-trial/, where we need to evaluate the token for the top-level site before the iframe is loaded.

Now we have achieved the behavior we want, and we will soon revise our code so that our authentication flow is not disrupted by this.

So, I consider this issue closed.
